packages/next/build/webpack/loaders/next-image-loader.js

```diff
@@ -0,0 +1,58 @@
+import loaderUtils from 'next/dist/compiled/loader-utils'
+import sizeOf from 'image-size'
+import { processBuffer } from '../../../next-server/server/lib/squoosh/main'
+import fs from 'fs'
+
+async function nextImageLoader(content) {
+  this.cacheable && this.cacheable(true)
+  this.addDependency(this.resourcePath)
+
+  const context = this.rootContext
+  const opts = { context, content }
+  const interpolatedName = loaderUtils.interpolateName(
+    this,
+    '/[path][name].[ext]',
+    opts
+  )
+
+  let extension = loaderUtils.interpolateName(this, '[ext]', opts)
+  if (extension === 'jpg') {
+    extension = 'jpeg'
+  }
+
+  if (interpolatedName.slice(0, 7) !== '/public') {
```

Comment:
Do we really want to enforce that? I feel it might be better to co-locate all resources with the components to which they belong.

```suggestion
  if (interpolatedName.slice(0, 8) !== '/public/') {
```

Reply:
Done. I took this check out altogether--you can now include an image from anywhere relative to the importing component.